{
 "cells": [
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "Loading Basic Data Analysis and Visualization Packages"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "Summary of Discussion With Nimrod\n",
    "\n",
    "exp: gene expression - each row is a gene \n",
    "patients: 06 - metastasis, 01 - primary\n",
    "numbers: counting molecules of a specific gene; [I think: if for gene X patient a has 100 and patient b 1, the molecules of the gene are 100 times more present than person a than in b]\n",
    "we should normalize these numbers: \n",
    "- perform log2(1+val); \n",
    "- this is normalized further by subtracting the avg (after log) and dividing by standard deviation\n",
    "\n",
    "mirna: micro rna\n",
    "each row is a micro rna\n",
    "we should use the same method for normalizing\n",
    "this data is collected in a different process (which is why it is separate) but the analysis is similar\n",
    "\n",
    "gtex data:\n",
    "- it looks normalized, but we should check if it actually is: check if std: 1 per row, mean 0\n",
    "- otherwise, normalize \n",
    "\n",
    "In all cases:\n",
    "- need to perform feature selection before analysis\n",
    "- then log2(1 + val)\n",
    "- remove features with low variance (because they are similar for all patients) (feature selection)\n",
    "- then remove mean and divide by std\n",
    "\n",
    "More Info\n",
    "In gtex: ENSG: ensemble ids, a type of id for genes\n",
    "In tcga: name of gene(first part).id: entree id (we should ignore those starting with X..)\n",
    "There are packages that can translate between the two types of ids! (look for: gene id conversions)\n",
    "But: there is no 100% agreement on what is a gene and what isn't so this won't be perfect\n",
    "\n",
    "Suggestions:\n",
    "- start with clustering of genes and of patients.\n",
    "- then do this for the primary and for metasteses (01, 06)\n",
    "- do pca for data of all samples\n",
    "- color data based on primary or metastesses\n",
    "- see how different these things are (easy with visualization)\n",
    "- find genes that behave differently, and microrna that behave differently\n",
    "- we would expect the micrornas that act differently will affect the genes. we should check these relationships with external database\n",
    "\n",
    "-start with clustering and visualizations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [],
   "source": [
    "#basic data analysis packages\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "\n",
    "#basic data visualization packages\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "import seaborn as sns"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "Loading Data From http://acgt.cs.tau.ac.il/multi_omic_benchmark/download.html (SKCM)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [],
   "source": [
    "# load exp, mirna data\n",
    "\n",
    "\"\"\"\n",
    "creating dict containing \n",
    "- filename (of data provided for the project)\n",
    "- values: [fist column name, dataset name, separator]\n",
    "\"\"\"\n",
    "\n",
    "data = {'exp':['gene', 'gene_data', ' '], \n",
    "        'mirna': ['micro_rna', 'micro_rna_data', ' '],\n",
    "        'gtex_data': ['ensemble_id', 'gtex_data', '\\t']}\n",
    "\n",
    "dataframes = {}\n",
    "\n",
    "\"\"\"\n",
    "load_dataframe will add relevant data to dataframes\n",
    "\"\"\"\n",
    "\n",
    "def load_dataframe(filename, values):\n",
    "    title_row = pd.read_csv('{}.csv'.format(filename), header=None, nrows=1) #read table column names as row\n",
    "    names = title_row.iloc[0, 0].replace('\"', '').split(values[2]) #create colum title list\n",
    "    if filename == 'gtex_data':\n",
    "        names[0] = values[0]\n",
    "    else:\n",
    "        names = [values[0]] + names\n",
    "    dataframes[values[1]] = pd.read_csv('{}.csv'.format(filename), names=names, skiprows=1, sep=values[2]) #read data    \n",
    "\n",
    "for x in data.keys():\n",
    "    load_dataframe(x, data[x])\n",
    " \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [],
   "source": [
    "# removing rows from 'exp' where genes begin in \"X..\" (following discussion with Nimrod, they are irrelevant)\n",
    "dataframes['gene_data'] = dataframes['gene_data'][~dataframes['gene_data'].gene.str.contains(\"X..\")] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [],
   "source": [
    "# load survival data\n",
    "\n",
    "\"\"\"\n",
    "survival data needs to be loaded in a different way (based on data observation)\n",
    "we will load it and add it to our dataframes dictionary (for consistency)\n",
    "\n",
    "the column named \"Death\" has been renamed as \"isDead\" for clarity\n",
    "\"\"\"\n",
    "survival = pd.read_csv('survival.csv', names=['PatientID', 'Survival', 'isDead'], skiprows=1, sep='\\t')\n",
    "\n",
    "dataframes['survival_data'] = survival"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "Familiarity With The Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>TCGA.3N.A9WB.06</th>\n",
       "      <th>TCGA.3N.A9WC.06</th>\n",
       "      <th>TCGA.3N.A9WD.06</th>\n",
       "      <th>TCGA.BF.A1PU.01</th>\n",
       "      <th>TCGA.BF.A1PV.01</th>\n",
       "      <th>TCGA.BF.A1PX.01</th>\n",
       "      <th>TCGA.BF.A1PZ.01</th>\n",
       "      <th>TCGA.BF.A1Q0.01</th>\n",
       "      <th>TCGA.BF.A3DJ.01</th>\n",
       "      <th>...</th>\n",
       "      <th>TCGA.XV.AB01.06</th>\n",
       "      <th>TCGA.YD.A89C.06</th>\n",
       "      <th>TCGA.YD.A9TA.06</th>\n",
       "      <th>TCGA.YD.A9TB.06</th>\n",
       "      <th>TCGA.YG.AA3N.01</th>\n",
       "      <th>TCGA.YG.AA3O.06</th>\n",
       "      <th>TCGA.YG.AA3P.06</th>\n",
       "      <th>TCGA.Z2.A8RT.06</th>\n",
       "      <th>TCGA.Z2.AA3S.06</th>\n",
       "      <th>TCGA.Z2.AA3V.06</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>A1BG.1</td>\n",
       "      <td>381.0662</td>\n",
       "      <td>195.1822</td>\n",
       "      <td>360.8794</td>\n",
       "      <td>176.3994</td>\n",
       "      <td>216.8470</td>\n",
       "      <td>285.0608</td>\n",
       "      <td>424.5848</td>\n",
       "      <td>327.8758</td>\n",
       "      <td>232.2936</td>\n",
       "      <td>...</td>\n",
       "      <td>428.9256</td>\n",
       "      <td>184.5297</td>\n",
       "      <td>88.4289</td>\n",
       "      <td>264.5184</td>\n",
       "      <td>238.8846</td>\n",
       "      <td>207.9831</td>\n",
       "      <td>33.5642</td>\n",
       "      <td>336.4070</td>\n",
       "      <td>349.4478</td>\n",
       "      <td>717.2209</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>A1CF.29974</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.7092</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>1.3482</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.3876</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>A2BP1.54715</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>6.3830</td>\n",
       "      <td>1.2987</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.5522</td>\n",
       "      <td>0.3380</td>\n",
       "      <td>66.0629</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.2880</td>\n",
       "      <td>0.6539</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>A2LD1.87769</td>\n",
       "      <td>250.1979</td>\n",
       "      <td>160.7548</td>\n",
       "      <td>97.1986</td>\n",
       "      <td>163.2338</td>\n",
       "      <td>60.8727</td>\n",
       "      <td>125.6900</td>\n",
       "      <td>78.8939</td>\n",
       "      <td>67.7049</td>\n",
       "      <td>92.9554</td>\n",
       "      <td>...</td>\n",
       "      <td>47.5207</td>\n",
       "      <td>102.4404</td>\n",
       "      <td>36.9476</td>\n",
       "      <td>146.3544</td>\n",
       "      <td>88.2441</td>\n",
       "      <td>86.2831</td>\n",
       "      <td>50.0426</td>\n",
       "      <td>37.3178</td>\n",
       "      <td>152.9786</td>\n",
       "      <td>50.7438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>A2ML1.144568</td>\n",
       "      <td>7.2698</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>7.7922</td>\n",
       "      <td>0.5977</td>\n",
       "      <td>3277.0247</td>\n",
       "      <td>26.8887</td>\n",
       "      <td>5.6324</td>\n",
       "      <td>984.6631</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>1701.2700</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.2880</td>\n",
       "      <td>0.0000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 474 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            gene  TCGA.3N.A9WB.06  TCGA.3N.A9WC.06  TCGA.3N.A9WD.06  \\\n",
       "29        A1BG.1         381.0662         195.1822         360.8794   \n",
       "30    A1CF.29974           0.0000           0.0000           0.7092   \n",
       "31   A2BP1.54715           0.0000           0.0000           6.3830   \n",
       "32   A2LD1.87769         250.1979         160.7548          97.1986   \n",
       "33  A2ML1.144568           7.2698           0.0000           0.0000   \n",
       "\n",
       "    TCGA.BF.A1PU.01  TCGA.BF.A1PV.01  TCGA.BF.A1PX.01  TCGA.BF.A1PZ.01  \\\n",
       "29         176.3994         216.8470         285.0608         424.5848   \n",
       "30           0.0000           0.0000           0.0000           0.0000   \n",
       "31           1.2987           0.0000           0.0000           0.0000   \n",
       "32         163.2338          60.8727         125.6900          78.8939   \n",
       "33           7.7922           0.5977        3277.0247          26.8887   \n",
       "\n",
       "    TCGA.BF.A1Q0.01  TCGA.BF.A3DJ.01  ...  TCGA.XV.AB01.06  TCGA.YD.A89C.06  \\\n",
       "29         327.8758         232.2936  ...         428.9256         184.5297   \n",
       "30           0.0000           0.0000  ...           0.0000           0.0000   \n",
       "31           0.0000           0.0000  ...           0.0000           0.0000   \n",
       "32          67.7049          92.9554  ...          47.5207         102.4404   \n",
       "33           5.6324         984.6631  ...           0.0000           0.0000   \n",
       "\n",
       "    TCGA.YD.A9TA.06  TCGA.YD.A9TB.06  TCGA.YG.AA3N.01  TCGA.YG.AA3O.06  \\\n",
       "29          88.4289         264.5184         238.8846         207.9831   \n",
       "30           1.3482           0.0000           0.0000           0.0000   \n",
       "31           0.0000           0.0000           0.5522           0.3380   \n",
       "32          36.9476         146.3544          88.2441          86.2831   \n",
       "33           0.0000           0.0000        1701.2700           0.0000   \n",
       "\n",
       "    TCGA.YG.AA3P.06  TCGA.Z2.A8RT.06  TCGA.Z2.AA3S.06  TCGA.Z2.AA3V.06  \n",
       "29          33.5642         336.4070         349.4478         717.2209  \n",
       "30           0.0000           0.3876           0.0000           0.0000  \n",
       "31          66.0629           0.0000           0.2880           0.6539  \n",
       "32          50.0426          37.3178         152.9786          50.7438  \n",
       "33           0.0000           0.0000           0.2880           0.0000  \n",
       "\n",
       "[5 rows x 474 columns]"
      ]
     },
     "execution_count": 108,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# sample exp data\n",
    "dataframes['gene_data'].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>micro_rna</th>\n",
       "      <th>TCGA.3N.A9WC.06</th>\n",
       "      <th>TCGA.3N.A9WD.06</th>\n",
       "      <th>TCGA.BF.A1PU.01</th>\n",
       "      <th>TCGA.BF.A1PV.01</th>\n",
       "      <th>TCGA.BF.A1PX.01</th>\n",
       "      <th>TCGA.BF.A1PZ.01</th>\n",
       "      <th>TCGA.BF.A1Q0.01</th>\n",
       "      <th>TCGA.BF.A3DJ.01</th>\n",
       "      <th>TCGA.BF.A3DL.01</th>\n",
       "      <th>...</th>\n",
       "      <th>TCGA.XV.AB01.06</th>\n",
       "      <th>TCGA.YD.A89C.06</th>\n",
       "      <th>TCGA.YD.A9TA.06</th>\n",
       "      <th>TCGA.YD.A9TB.06</th>\n",
       "      <th>TCGA.YG.AA3N.01</th>\n",
       "      <th>TCGA.YG.AA3O.06</th>\n",
       "      <th>TCGA.YG.AA3P.06</th>\n",
       "      <th>TCGA.Z2.A8RT.06</th>\n",
       "      <th>TCGA.Z2.AA3S.06</th>\n",
       "      <th>TCGA.Z2.AA3V.06</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>hsa.let.7a.1</td>\n",
       "      <td>14676.834734</td>\n",
       "      <td>25230.440999</td>\n",
       "      <td>23563.756602</td>\n",
       "      <td>14330.771178</td>\n",
       "      <td>13740.462752</td>\n",
       "      <td>16199.355403</td>\n",
       "      <td>10114.610880</td>\n",
       "      <td>16712.764049</td>\n",
       "      <td>35403.380723</td>\n",
       "      <td>...</td>\n",
       "      <td>10317.628480</td>\n",
       "      <td>4797.673951</td>\n",
       "      <td>9765.591431</td>\n",
       "      <td>7878.460725</td>\n",
       "      <td>13157.578121</td>\n",
       "      <td>12493.132639</td>\n",
       "      <td>23455.671756</td>\n",
       "      <td>10805.087024</td>\n",
       "      <td>12216.751081</td>\n",
       "      <td>24025.124723</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>hsa.let.7a.2</td>\n",
       "      <td>29175.042697</td>\n",
       "      <td>50426.377848</td>\n",
       "      <td>47180.136901</td>\n",
       "      <td>28590.374972</td>\n",
       "      <td>27700.049991</td>\n",
       "      <td>32326.559988</td>\n",
       "      <td>20268.091283</td>\n",
       "      <td>33860.575382</td>\n",
       "      <td>70851.588875</td>\n",
       "      <td>...</td>\n",
       "      <td>20559.441088</td>\n",
       "      <td>9682.578337</td>\n",
       "      <td>19545.557812</td>\n",
       "      <td>15938.581844</td>\n",
       "      <td>26378.793396</td>\n",
       "      <td>24907.572969</td>\n",
       "      <td>46879.348693</td>\n",
       "      <td>21513.398318</td>\n",
       "      <td>24460.759020</td>\n",
       "      <td>47953.696317</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>hsa.let.7a.3</td>\n",
       "      <td>14729.348617</td>\n",
       "      <td>25459.749831</td>\n",
       "      <td>23733.441179</td>\n",
       "      <td>14277.736854</td>\n",
       "      <td>13996.431180</td>\n",
       "      <td>16282.583317</td>\n",
       "      <td>10174.933610</td>\n",
       "      <td>17208.997960</td>\n",
       "      <td>35473.928808</td>\n",
       "      <td>...</td>\n",
       "      <td>10285.834727</td>\n",
       "      <td>4897.314490</td>\n",
       "      <td>9836.841186</td>\n",
       "      <td>8034.707489</td>\n",
       "      <td>13248.062199</td>\n",
       "      <td>12403.510843</td>\n",
       "      <td>23803.374490</td>\n",
       "      <td>10905.788400</td>\n",
       "      <td>12680.117674</td>\n",
       "      <td>24210.389886</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>hsa.let.7b</td>\n",
       "      <td>15377.019845</td>\n",
       "      <td>34387.698695</td>\n",
       "      <td>54133.369009</td>\n",
       "      <td>14988.630767</td>\n",
       "      <td>49402.682161</td>\n",
       "      <td>15234.151107</td>\n",
       "      <td>8981.281730</td>\n",
       "      <td>37588.292821</td>\n",
       "      <td>73681.595915</td>\n",
       "      <td>...</td>\n",
       "      <td>12514.954736</td>\n",
       "      <td>1542.629784</td>\n",
       "      <td>8806.219729</td>\n",
       "      <td>13989.262271</td>\n",
       "      <td>11230.764427</td>\n",
       "      <td>14622.196769</td>\n",
       "      <td>10265.711191</td>\n",
       "      <td>8431.948143</td>\n",
       "      <td>1860.659154</td>\n",
       "      <td>20967.711434</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>hsa.let.7c</td>\n",
       "      <td>306.330986</td>\n",
       "      <td>10490.699334</td>\n",
       "      <td>1727.530141</td>\n",
       "      <td>1659.272408</td>\n",
       "      <td>3080.541241</td>\n",
       "      <td>1126.121575</td>\n",
       "      <td>848.324435</td>\n",
       "      <td>255.635651</td>\n",
       "      <td>3177.603335</td>\n",
       "      <td>...</td>\n",
       "      <td>339.503915</td>\n",
       "      <td>96.762834</td>\n",
       "      <td>502.810772</td>\n",
       "      <td>2268.166510</td>\n",
       "      <td>2037.979847</td>\n",
       "      <td>528.987186</td>\n",
       "      <td>10879.780431</td>\n",
       "      <td>451.620072</td>\n",
       "      <td>1644.875692</td>\n",
       "      <td>1120.047059</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 453 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      micro_rna  TCGA.3N.A9WC.06  TCGA.3N.A9WD.06  TCGA.BF.A1PU.01  \\\n",
       "0  hsa.let.7a.1     14676.834734     25230.440999     23563.756602   \n",
       "1  hsa.let.7a.2     29175.042697     50426.377848     47180.136901   \n",
       "2  hsa.let.7a.3     14729.348617     25459.749831     23733.441179   \n",
       "3    hsa.let.7b     15377.019845     34387.698695     54133.369009   \n",
       "4    hsa.let.7c       306.330986     10490.699334      1727.530141   \n",
       "\n",
       "   TCGA.BF.A1PV.01  TCGA.BF.A1PX.01  TCGA.BF.A1PZ.01  TCGA.BF.A1Q0.01  \\\n",
       "0     14330.771178     13740.462752     16199.355403     10114.610880   \n",
       "1     28590.374972     27700.049991     32326.559988     20268.091283   \n",
       "2     14277.736854     13996.431180     16282.583317     10174.933610   \n",
       "3     14988.630767     49402.682161     15234.151107      8981.281730   \n",
       "4      1659.272408      3080.541241      1126.121575       848.324435   \n",
       "\n",
       "   TCGA.BF.A3DJ.01  TCGA.BF.A3DL.01  ...  TCGA.XV.AB01.06  TCGA.YD.A89C.06  \\\n",
       "0     16712.764049     35403.380723  ...     10317.628480      4797.673951   \n",
       "1     33860.575382     70851.588875  ...     20559.441088      9682.578337   \n",
       "2     17208.997960     35473.928808  ...     10285.834727      4897.314490   \n",
       "3     37588.292821     73681.595915  ...     12514.954736      1542.629784   \n",
       "4       255.635651      3177.603335  ...       339.503915        96.762834   \n",
       "\n",
       "   TCGA.YD.A9TA.06  TCGA.YD.A9TB.06  TCGA.YG.AA3N.01  TCGA.YG.AA3O.06  \\\n",
       "0      9765.591431      7878.460725     13157.578121     12493.132639   \n",
       "1     19545.557812     15938.581844     26378.793396     24907.572969   \n",
       "2      9836.841186      8034.707489     13248.062199     12403.510843   \n",
       "3      8806.219729     13989.262271     11230.764427     14622.196769   \n",
       "4       502.810772      2268.166510      2037.979847       528.987186   \n",
       "\n",
       "   TCGA.YG.AA3P.06  TCGA.Z2.A8RT.06  TCGA.Z2.AA3S.06  TCGA.Z2.AA3V.06  \n",
       "0     23455.671756     10805.087024     12216.751081     24025.124723  \n",
       "1     46879.348693     21513.398318     24460.759020     47953.696317  \n",
       "2     23803.374490     10905.788400     12680.117674     24210.389886  \n",
       "3     10265.711191      8431.948143      1860.659154     20967.711434  \n",
       "4     10879.780431       451.620072      1644.875692      1120.047059  \n",
       "\n",
       "[5 rows x 453 columns]"
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# sample mirna data\n",
    "dataframes['micro_rna_data'].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ensemble_id</th>\n",
       "      <th>GTEX-111YS-0006-SM-5NQBE</th>\n",
       "      <th>GTEX-1122O-0005-SM-5O99J</th>\n",
       "      <th>GTEX-1128S-0005-SM-5P9HI</th>\n",
       "      <th>GTEX-113IC-0006-SM-5NQ9C</th>\n",
       "      <th>GTEX-113JC-0006-SM-5O997</th>\n",
       "      <th>GTEX-117XS-0005-SM-5PNU6</th>\n",
       "      <th>GTEX-117YW-0005-SM-5NQ8Z</th>\n",
       "      <th>GTEX-1192W-0005-SM-5NQBQ</th>\n",
       "      <th>GTEX-11DXW-0006-SM-5NQ7Y</th>\n",
       "      <th>...</th>\n",
       "      <th>GTEX-ZVE2-0006-SM-51MRW</th>\n",
       "      <th>GTEX-ZVP2-0005-SM-51MRK</th>\n",
       "      <th>GTEX-ZVT2-0005-SM-57WBW</th>\n",
       "      <th>GTEX-ZVT3-0006-SM-51MT9</th>\n",
       "      <th>GTEX-ZVT4-0006-SM-57WB8</th>\n",
       "      <th>GTEX-ZVTK-0006-SM-57WBK</th>\n",
       "      <th>GTEX-ZVZP-0006-SM-51MSW</th>\n",
       "      <th>GTEX-ZVZQ-0006-SM-51MR8</th>\n",
       "      <th>GTEX-ZXES-0005-SM-57WCB</th>\n",
       "      <th>GTEX-ZXG5-0005-SM-57WCN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>\"ENSG00000227232\"</td>\n",
       "      <td>-0.169609</td>\n",
       "      <td>0.054703</td>\n",
       "      <td>0.009834</td>\n",
       "      <td>0.062832</td>\n",
       "      <td>-0.111598</td>\n",
       "      <td>0.013749</td>\n",
       "      <td>0.059378</td>\n",
       "      <td>-0.103492</td>\n",
       "      <td>-0.056906</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.135461</td>\n",
       "      <td>0.031812</td>\n",
       "      <td>-0.033919</td>\n",
       "      <td>-0.083926</td>\n",
       "      <td>0.057796</td>\n",
       "      <td>-0.026684</td>\n",
       "      <td>0.076550</td>\n",
       "      <td>0.074503</td>\n",
       "      <td>-0.024322</td>\n",
       "      <td>0.008076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>\"ENSG00000238009\"</td>\n",
       "      <td>-0.268123</td>\n",
       "      <td>0.129532</td>\n",
       "      <td>-0.226318</td>\n",
       "      <td>-0.594374</td>\n",
       "      <td>-0.247985</td>\n",
       "      <td>0.440750</td>\n",
       "      <td>0.012203</td>\n",
       "      <td>0.023756</td>\n",
       "      <td>0.023010</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.233336</td>\n",
       "      <td>-0.041431</td>\n",
       "      <td>0.150914</td>\n",
       "      <td>0.142757</td>\n",
       "      <td>0.181792</td>\n",
       "      <td>-0.067478</td>\n",
       "      <td>0.196848</td>\n",
       "      <td>0.086229</td>\n",
       "      <td>-0.274816</td>\n",
       "      <td>-0.030805</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>\"ENSG00000233750\"</td>\n",
       "      <td>-0.168988</td>\n",
       "      <td>-0.183577</td>\n",
       "      <td>-0.164673</td>\n",
       "      <td>-0.587763</td>\n",
       "      <td>-0.170246</td>\n",
       "      <td>0.177095</td>\n",
       "      <td>0.092399</td>\n",
       "      <td>0.186742</td>\n",
       "      <td>0.050464</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.647793</td>\n",
       "      <td>-0.179416</td>\n",
       "      <td>-0.039154</td>\n",
       "      <td>-0.157434</td>\n",
       "      <td>-0.072724</td>\n",
       "      <td>0.008101</td>\n",
       "      <td>0.326528</td>\n",
       "      <td>-0.103061</td>\n",
       "      <td>-0.018962</td>\n",
       "      <td>-0.099040</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>\"ENSG00000237683\"</td>\n",
       "      <td>-0.015901</td>\n",
       "      <td>-0.114264</td>\n",
       "      <td>0.273116</td>\n",
       "      <td>-0.442626</td>\n",
       "      <td>-0.249558</td>\n",
       "      <td>0.040346</td>\n",
       "      <td>0.092957</td>\n",
       "      <td>0.060168</td>\n",
       "      <td>0.024587</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.200647</td>\n",
       "      <td>-0.036321</td>\n",
       "      <td>0.093315</td>\n",
       "      <td>-0.097231</td>\n",
       "      <td>0.055875</td>\n",
       "      <td>-0.226099</td>\n",
       "      <td>-0.198103</td>\n",
       "      <td>-0.151890</td>\n",
       "      <td>-0.091329</td>\n",
       "      <td>-0.023483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>\"ENSG00000268903\"</td>\n",
       "      <td>0.033084</td>\n",
       "      <td>-0.136116</td>\n",
       "      <td>0.357690</td>\n",
       "      <td>-0.209781</td>\n",
       "      <td>-1.040812</td>\n",
       "      <td>0.290923</td>\n",
       "      <td>-0.224477</td>\n",
       "      <td>-0.395459</td>\n",
       "      <td>0.080350</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.395148</td>\n",
       "      <td>0.019980</td>\n",
       "      <td>0.151770</td>\n",
       "      <td>0.050996</td>\n",
       "      <td>0.102515</td>\n",
       "      <td>-0.360847</td>\n",
       "      <td>-0.076868</td>\n",
       "      <td>0.020539</td>\n",
       "      <td>-0.067300</td>\n",
       "      <td>0.040908</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 394 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         ensemble_id  GTEX-111YS-0006-SM-5NQBE  GTEX-1122O-0005-SM-5O99J  \\\n",
       "0  \"ENSG00000227232\"                 -0.169609                  0.054703   \n",
       "1  \"ENSG00000238009\"                 -0.268123                  0.129532   \n",
       "2  \"ENSG00000233750\"                 -0.168988                 -0.183577   \n",
       "3  \"ENSG00000237683\"                 -0.015901                 -0.114264   \n",
       "4  \"ENSG00000268903\"                  0.033084                 -0.136116   \n",
       "\n",
       "   GTEX-1128S-0005-SM-5P9HI  GTEX-113IC-0006-SM-5NQ9C  \\\n",
       "0                  0.009834                  0.062832   \n",
       "1                 -0.226318                 -0.594374   \n",
       "2                 -0.164673                 -0.587763   \n",
       "3                  0.273116                 -0.442626   \n",
       "4                  0.357690                 -0.209781   \n",
       "\n",
       "   GTEX-113JC-0006-SM-5O997  GTEX-117XS-0005-SM-5PNU6  \\\n",
       "0                 -0.111598                  0.013749   \n",
       "1                 -0.247985                  0.440750   \n",
       "2                 -0.170246                  0.177095   \n",
       "3                 -0.249558                  0.040346   \n",
       "4                 -1.040812                  0.290923   \n",
       "\n",
       "   GTEX-117YW-0005-SM-5NQ8Z  GTEX-1192W-0005-SM-5NQBQ  \\\n",
       "0                  0.059378                 -0.103492   \n",
       "1                  0.012203                  0.023756   \n",
       "2                  0.092399                  0.186742   \n",
       "3                  0.092957                  0.060168   \n",
       "4                 -0.224477                 -0.395459   \n",
       "\n",
       "   GTEX-11DXW-0006-SM-5NQ7Y  ...  GTEX-ZVE2-0006-SM-51MRW  \\\n",
       "0                 -0.056906  ...                -0.135461   \n",
       "1                  0.023010  ...                -0.233336   \n",
       "2                  0.050464  ...                -0.647793   \n",
       "3                  0.024587  ...                -0.200647   \n",
       "4                  0.080350  ...                -0.395148   \n",
       "\n",
       "   GTEX-ZVP2-0005-SM-51MRK  GTEX-ZVT2-0005-SM-57WBW  GTEX-ZVT3-0006-SM-51MT9  \\\n",
       "0                 0.031812                -0.033919                -0.083926   \n",
       "1                -0.041431                 0.150914                 0.142757   \n",
       "2                -0.179416                -0.039154                -0.157434   \n",
       "3                -0.036321                 0.093315                -0.097231   \n",
       "4                 0.019980                 0.151770                 0.050996   \n",
       "\n",
       "   GTEX-ZVT4-0006-SM-57WB8  GTEX-ZVTK-0006-SM-57WBK  GTEX-ZVZP-0006-SM-51MSW  \\\n",
       "0                 0.057796                -0.026684                 0.076550   \n",
       "1                 0.181792                -0.067478                 0.196848   \n",
       "2                -0.072724                 0.008101                 0.326528   \n",
       "3                 0.055875                -0.226099                -0.198103   \n",
       "4                 0.102515                -0.360847                -0.076868   \n",
       "\n",
       "   GTEX-ZVZQ-0006-SM-51MR8  GTEX-ZXES-0005-SM-57WCB  GTEX-ZXG5-0005-SM-57WCN  \n",
       "0                 0.074503                -0.024322                 0.008076  \n",
       "1                 0.086229                -0.274816                -0.030805  \n",
       "2                -0.103061                -0.018962                -0.099040  \n",
       "3                -0.151890                -0.091329                -0.023483  \n",
       "4                 0.020539                -0.067300                 0.040908  \n",
       "\n",
       "[5 rows x 394 columns]"
      ]
     },
     "execution_count": 110,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dataframes['gtex_data'].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PatientID</th>\n",
       "      <th>Survival</th>\n",
       "      <th>isDead</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-3N-A9WB-06</td>\n",
       "      <td>518.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCGA-3N-A9WC-06</td>\n",
       "      <td>2022.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TCGA-3N-A9WD-06</td>\n",
       "      <td>395.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCGA-BF-A1PU-01</td>\n",
       "      <td>387.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TCGA-BF-A1PV-01</td>\n",
       "      <td>14.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         PatientID  Survival  isDead\n",
       "0  TCGA-3N-A9WB-06     518.0     1.0\n",
       "1  TCGA-3N-A9WC-06    2022.0     0.0\n",
       "2  TCGA-3N-A9WD-06     395.0     1.0\n",
       "3  TCGA-BF-A1PU-01     387.0     0.0\n",
       "4  TCGA-BF-A1PV-01      14.0     0.0"
      ]
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# sample survival data\n",
    "dataframes['survival_data'].head()"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "Loading Data From http://acgt.cs.tau.ac.il/multi_omic_benchmark/download.html (clinical data -> melanoma)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [],
   "source": [
    "# load clinical data data\n",
    "\n",
    "\"\"\"\n",
    "the data has a problem in line 38, 39:\n",
    "they are series with 3 elements (total values: 105, unlike total 103 in other rows)\n",
    "we discluded this data\n",
    "\"\"\"\n",
    "\n",
    "clinical_titles = pd.read_csv('melanoma.txt', header=None, nrows=1) \n",
    "clinical_data_titles = clinical_titles.iloc[0, 0].split('\\t')\n",
    "\n",
    "clinical_data = pd.read_csv('melanoma.txt', header=None, names=clinical_data_titles, skiprows=[0, 38,39], sep='\\t') \n",
    "\n",
    "dataframes['clinical_data'] = clinical_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sampleID</th>\n",
       "      <th>_EVENT</th>\n",
       "      <th>_INTEGRATION</th>\n",
       "      <th>_OS</th>\n",
       "      <th>_OS_IND</th>\n",
       "      <th>_OS_UNIT</th>\n",
       "      <th>_PATIENT</th>\n",
       "      <th>_RFS</th>\n",
       "      <th>_RFS_IND</th>\n",
       "      <th>_RFS_UNIT</th>\n",
       "      <th>...</th>\n",
       "      <th>_GENOMIC_ID_data/public/TCGA/SKCM/miRNA_HiSeq_gene</th>\n",
       "      <th>_GENOMIC_ID_TCGA_SKCM_RPPA</th>\n",
       "      <th>_GENOMIC_ID_TCGA_SKCM_mutation_bcm_gene</th>\n",
       "      <th>_GENOMIC_ID_TCGA_SKCM_mutation_broad_gene</th>\n",
       "      <th>_GENOMIC_ID_TCGA_SKCM_gistic2</th>\n",
       "      <th>_GENOMIC_ID_TCGA_SKCM_mutation</th>\n",
       "      <th>_GENOMIC_ID_TCGA_SKCM_exp_HiSeqV2_exon</th>\n",
       "      <th>_GENOMIC_ID_TCGA_SKCM_PDMRNAseqCNV</th>\n",
       "      <th>_GENOMIC_ID_TCGA_SKCM_PDMRNAseq</th>\n",
       "      <th>_GENOMIC_ID_TCGA_SKCM_exp_HiSeqV2_percentile</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-3N-A9WB-06</td>\n",
       "      <td>1.0</td>\n",
       "      <td>TCGA-3N-A9WB-06</td>\n",
       "      <td>518.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>days</td>\n",
       "      <td>TCGA-3N-A9WB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>days</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-3N-A9WB-06A-11D-A38G-08</td>\n",
       "      <td>TCGA-3N-A9WB-06A-11D-A38F-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4c243ea9-dfe1-42f0-a887-3c901fb38542</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4c243ea9-dfe1-42f0-a887-3c901fb38542</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCGA-3N-A9WC-06</td>\n",
       "      <td>0.0</td>\n",
       "      <td>TCGA-3N-A9WC-06</td>\n",
       "      <td>2022.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>days</td>\n",
       "      <td>TCGA-3N-A9WC</td>\n",
       "      <td>2022.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>days</td>\n",
       "      <td>...</td>\n",
       "      <td>TCGA-3N-A9WC-06</td>\n",
       "      <td>F9B0B31B-24AB-4F30-8EF8-4A4ED282EF04</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-3N-A9WC-06A-11D-A38G-08</td>\n",
       "      <td>TCGA-3N-A9WC-06A-11D-A38F-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>a64ae1f5-a189-4173-be13-903bd7637869</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>a64ae1f5-a189-4173-be13-903bd7637869</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TCGA-3N-A9WD-06</td>\n",
       "      <td>1.0</td>\n",
       "      <td>TCGA-3N-A9WD-06</td>\n",
       "      <td>395.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>days</td>\n",
       "      <td>TCGA-3N-A9WD</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>days</td>\n",
       "      <td>...</td>\n",
       "      <td>TCGA-3N-A9WD-06</td>\n",
       "      <td>6AFC3678-9499-49A6-AD7A-30F3D185B4BD</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-3N-A9WD-06A-11D-A38G-08</td>\n",
       "      <td>TCGA-3N-A9WD-06A-11D-A38F-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ac19f7cf-670b-4dcc-a26b-db0f56377231</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ac19f7cf-670b-4dcc-a26b-db0f56377231</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCGA-BF-A1PU-01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>TCGA-BF-A1PU-01</td>\n",
       "      <td>387.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>days</td>\n",
       "      <td>TCGA-BF-A1PU</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>days</td>\n",
       "      <td>...</td>\n",
       "      <td>TCGA-BF-A1PU-01</td>\n",
       "      <td>7E74CAFD-C47B-4F98-A349-BBD2AC699D2D</td>\n",
       "      <td>TCGA-BF-A1PU-01A-11D-A19A-08</td>\n",
       "      <td>TCGA-BF-A1PU-01A-11D-A19A-08</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-BF-A1PU-01A-11D-A19A-08</td>\n",
       "      <td>9ff4d588-da73-4e26-975d-2b098fe74984</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9ff4d588-da73-4e26-975d-2b098fe74984</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TCGA-BF-A1PV-01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>TCGA-BF-A1PV-01</td>\n",
       "      <td>14.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>days</td>\n",
       "      <td>TCGA-BF-A1PV</td>\n",
       "      <td>14.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>days</td>\n",
       "      <td>...</td>\n",
       "      <td>TCGA-BF-A1PV-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-BF-A1PV-01A-11D-A19A-08</td>\n",
       "      <td>TCGA-BF-A1PV-01A-11D-A19A-08</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-BF-A1PV-01A-11D-A19A-08</td>\n",
       "      <td>a57841bb-fa68-4d84-85e0-2d5e4f3ce828</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>a57841bb-fa68-4d84-85e0-2d5e4f3ce828</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 103 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          sampleID  _EVENT     _INTEGRATION     _OS  _OS_IND _OS_UNIT  \\\n",
       "0  TCGA-3N-A9WB-06     1.0  TCGA-3N-A9WB-06   518.0      1.0     days   \n",
       "1  TCGA-3N-A9WC-06     0.0  TCGA-3N-A9WC-06  2022.0      0.0     days   \n",
       "2  TCGA-3N-A9WD-06     1.0  TCGA-3N-A9WD-06   395.0      1.0     days   \n",
       "3  TCGA-BF-A1PU-01     0.0  TCGA-BF-A1PU-01   387.0      0.0     days   \n",
       "4  TCGA-BF-A1PV-01     0.0  TCGA-BF-A1PV-01    14.0      0.0     days   \n",
       "\n",
       "       _PATIENT    _RFS  _RFS_IND _RFS_UNIT  ...  \\\n",
       "0  TCGA-3N-A9WB     NaN       NaN      days  ...   \n",
       "1  TCGA-3N-A9WC  2022.0       0.0      days  ...   \n",
       "2  TCGA-3N-A9WD     NaN       NaN      days  ...   \n",
       "3  TCGA-BF-A1PU     NaN       NaN      days  ...   \n",
       "4  TCGA-BF-A1PV    14.0       0.0      days  ...   \n",
       "\n",
       "   _GENOMIC_ID_data/public/TCGA/SKCM/miRNA_HiSeq_gene  \\\n",
       "0                                                NaN    \n",
       "1                                    TCGA-3N-A9WC-06    \n",
       "2                                    TCGA-3N-A9WD-06    \n",
       "3                                    TCGA-BF-A1PU-01    \n",
       "4                                    TCGA-BF-A1PV-01    \n",
       "\n",
       "             _GENOMIC_ID_TCGA_SKCM_RPPA  \\\n",
       "0                                   NaN   \n",
       "1  F9B0B31B-24AB-4F30-8EF8-4A4ED282EF04   \n",
       "2  6AFC3678-9499-49A6-AD7A-30F3D185B4BD   \n",
       "3  7E74CAFD-C47B-4F98-A349-BBD2AC699D2D   \n",
       "4                                   NaN   \n",
       "\n",
       "  _GENOMIC_ID_TCGA_SKCM_mutation_bcm_gene  \\\n",
       "0                                     NaN   \n",
       "1                                     NaN   \n",
       "2                                     NaN   \n",
       "3            TCGA-BF-A1PU-01A-11D-A19A-08   \n",
       "4            TCGA-BF-A1PV-01A-11D-A19A-08   \n",
       "\n",
       "  _GENOMIC_ID_TCGA_SKCM_mutation_broad_gene _GENOMIC_ID_TCGA_SKCM_gistic2  \\\n",
       "0              TCGA-3N-A9WB-06A-11D-A38G-08  TCGA-3N-A9WB-06A-11D-A38F-01   \n",
       "1              TCGA-3N-A9WC-06A-11D-A38G-08  TCGA-3N-A9WC-06A-11D-A38F-01   \n",
       "2              TCGA-3N-A9WD-06A-11D-A38G-08  TCGA-3N-A9WD-06A-11D-A38F-01   \n",
       "3              TCGA-BF-A1PU-01A-11D-A19A-08                           NaN   \n",
       "4              TCGA-BF-A1PV-01A-11D-A19A-08                           NaN   \n",
       "\n",
       "  _GENOMIC_ID_TCGA_SKCM_mutation _GENOMIC_ID_TCGA_SKCM_exp_HiSeqV2_exon  \\\n",
       "0                            NaN   4c243ea9-dfe1-42f0-a887-3c901fb38542   \n",
       "1                            NaN   a64ae1f5-a189-4173-be13-903bd7637869   \n",
       "2                            NaN   ac19f7cf-670b-4dcc-a26b-db0f56377231   \n",
       "3   TCGA-BF-A1PU-01A-11D-A19A-08   9ff4d588-da73-4e26-975d-2b098fe74984   \n",
       "4   TCGA-BF-A1PV-01A-11D-A19A-08   a57841bb-fa68-4d84-85e0-2d5e4f3ce828   \n",
       "\n",
       "   _GENOMIC_ID_TCGA_SKCM_PDMRNAseqCNV _GENOMIC_ID_TCGA_SKCM_PDMRNAseq  \\\n",
       "0                                 NaN                             NaN   \n",
       "1                                 NaN                             NaN   \n",
       "2                                 NaN                             NaN   \n",
       "3                                 NaN                             NaN   \n",
       "4                                 NaN                             NaN   \n",
       "\n",
       "  _GENOMIC_ID_TCGA_SKCM_exp_HiSeqV2_percentile  \n",
       "0         4c243ea9-dfe1-42f0-a887-3c901fb38542  \n",
       "1         a64ae1f5-a189-4173-be13-903bd7637869  \n",
       "2         ac19f7cf-670b-4dcc-a26b-db0f56377231  \n",
       "3         9ff4d588-da73-4e26-975d-2b098fe74984  \n",
       "4         a57841bb-fa68-4d84-85e0-2d5e4f3ce828  \n",
       "\n",
       "[5 rows x 103 columns]"
      ]
     },
     "execution_count": 113,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#sample clinical data\n",
    "dataframes['clinical_data'].head()"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
