{
 "cells": [
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "Loading Basic Data Analysis and Visualization Packages"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "Summary of Discussion With Nimrod\n",
    "\n",
    "exp: gene expression - each row is a gene \n",
    "patients: 06 - metastasis, 01 - primary\n",
    "numbers: counting molecules of a specific gene; [I think: if for gene X patient a has 100 and patient b 1, the molecules of the gene are 100 times more present than person a than in b]\n",
    "we should normalize these numbers: \n",
    "- perform log2(1+val); \n",
    "- this is normalized further by subtracting the avg (after log) and dividing by standard deviation\n",
    "\n",
    "mirna: micro rna\n",
    "each row is a micro rna\n",
    "we should use the same method for normalizing\n",
    "this data is collected in a different process (which is why it is separate) but the analysis is similar\n",
    "\n",
    "gtex data:\n",
    "- it looks normalized, but we should check if it actually is: check if std: 1 per row, mean 0\n",
    "- otherwise, normalize \n",
    "\n",
    "In all cases:\n",
    "- need to perform feature selection before analysis\n",
    "- then log2(1 + val)\n",
    "- remove features with low variance (because they are similar for all patients) (feature selection)\n",
    "- then remove mean and divide by std\n",
    "\n",
    "More Info\n",
    "In gtex: ENSG: ensemble ids, a type of id for genes\n",
    "In tcga: name of gene(first part).id: entree id (we should ignore those starting with X..)\n",
    "There are packages that can translate between the two types of ids! (look for: gene id conversions)\n",
    "But: there is no 100% agreement on what is a gene and what isn't so this won't be perfect\n",
    "\n",
    "Suggestions:\n",
    "- start with clustering of genes and of patients.\n",
    "- then do this for the primary and for metasteses (01, 06)\n",
    "- do pca for data of all samples\n",
    "- color data based on primary or metastesses\n",
    "- see how different these things are (easy with visualization)\n",
    "- find genes that behave differently, and microrna that behave differently\n",
    "- we would expect the micrornas that act differently will affect the genes. we should check these relationships with external database\n",
    "\n",
    "-start with clustering and visualizations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [],
   "source": [
    "#basic data analysis packages\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "\n",
    "#basic data visualization packages\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "import seaborn as sns"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "Loading Data From http://acgt.cs.tau.ac.il/multi_omic_benchmark/download.html (SKCM)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [],
   "source": [
    "# load exp, mirna data\n",
    "\n",
    "\"\"\"\n",
    "creating dict containing \n",
    "- filename (of data provided for the project)\n",
    "- values: [fist column name, dataset name, separator]\n",
    "\"\"\"\n",
    "\n",
    "data = {'exp':['gene', 'gene_data', ' '], \n",
    "        'mirna': ['micro_rna', 'micro_rna_data', ' '],\n",
    "        'gtex_data': ['ensemble_id', 'gtex_data', '\\t']}\n",
    "\n",
    "dataframes = {}\n",
    "\n",
    "\"\"\"\n",
    "load_dataframe will add relevant data to dataframes\n",
    "\"\"\"\n",
    "\n",
    "def load_dataframe(filename, values):\n",
    "    title_row = pd.read_csv('{}.csv'.format(filename), header=None, nrows=1) #read table column names as row\n",
    "    names = title_row.iloc[0, 0].replace('\"', '').split(values[2]) #create colum title list\n",
    "    if filename == 'gtex_data':\n",
    "        names[0] = values[0]\n",
    "    else:\n",
    "        names = [values[0]] + names\n",
    "    dataframes[values[1]] = pd.read_csv('{}.csv'.format(filename), names=names, skiprows=1, sep=values[2]) #read data    \n",
    "\n",
    "for x in data.keys():\n",
    "    load_dataframe(x, data[x])\n",
    " \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [],
   "source": [
    "# removing rows from 'exp' where genes begin in \"X..\" (following discussion with Nimrod, they are irrelevant)\n",
    "dataframes['gene_data'] = dataframes['gene_data'][~dataframes['gene_data'].gene.str.contains(\"X..\")] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [],
   "source": [
    "# load survival data\n",
    "\n",
    "\"\"\"\n",
    "survival data needs to be loaded in a different way (based on data observation)\n",
    "we will load it and add it to our dataframes dictionary (for consistency)\n",
    "\n",
    "the column named \"Death\" has been renamed as \"isDead\" for clarity\n",
    "\"\"\"\n",
    "survival = pd.read_csv('survival.csv', names=['PatientID', 'Survival', 'isDead'], skiprows=1, sep='\\t')\n",
    "\n",
    "dataframes['survival_data'] = survival"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "Normalize The Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [],
   "source": [
    "def normalize_data(dataframe, column_name):\n",
    "    # perform log2 (val+1)\n",
    "    dataframe = dataframe.apply(lambda x: np.log2(x + 1) if x.name != column_name else x)\n",
    "    # find mean and std per row\n",
    "    mean_vals = dataframe.mean(axis=1)\n",
    "    std_vals = dataframe.std(axis=1)\n",
    "    # subtract means and divide by std\n",
    "    dataframe.iloc[:, 1:] = dataframe.iloc[:, 1:].subtract(mean_vals, axis=0).divide(std_vals, axis=0)\n",
    "    return dataframe\n",
    "\n",
    "dataframes['gene_data'] = normalize_data(dataframes['gene_data'], 'gene')\n",
    "dataframes['micro_rna_data'] = normalize_data(dataframes['micro_rna_data'], 'micro_rna')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['A1BG.1', 'A1CF.29974', 'A2BP1.54715', 'A2LD1.87769', 'A2ML1.144568',\n",
       "       'A2M.2', 'A4GALT.53947', 'A4GNT.51146', 'AAA1.404744', 'AAAS.8086',\n",
       "       ...\n",
       "       'ZUFSP.221302', 'ZW10.9183', 'ZWILCH.55055', 'ZWINT.11130',\n",
       "       'ZYG11A.440590', 'ZYG11B.79699', 'ZZEF1.23140', 'ZZZ3.26009',\n",
       "       'psiTPTE22.387590', 'tAKR.389932'],\n",
       "      dtype='object', name='gene', length=19363)"
      ]
     },
     "execution_count": 148,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# transpose gene data, micro rna data\n",
    "\n",
    "#transpose_data = [dataframes['gene_data'], dataframes['micro_rna_data']]\n",
    "\n",
    "a = dataframes['gene_data'].T # transpose\n",
    "a.columns = a.iloc[0, :] # use first row as column names\n",
    "a = a.iloc[1:, :]\n",
    "a.columns\n",
    "# a.rename(columns={\" gene \": \"patient\"})\n",
    "\n"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "Familiarity With The Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>29</th>\n",
       "      <th>30</th>\n",
       "      <th>31</th>\n",
       "      <th>32</th>\n",
       "      <th>33</th>\n",
       "      <th>34</th>\n",
       "      <th>35</th>\n",
       "      <th>36</th>\n",
       "      <th>37</th>\n",
       "      <th>38</th>\n",
       "      <th>...</th>\n",
       "      <th>20517</th>\n",
       "      <th>20518</th>\n",
       "      <th>20519</th>\n",
       "      <th>20520</th>\n",
       "      <th>20524</th>\n",
       "      <th>20525</th>\n",
       "      <th>20527</th>\n",
       "      <th>20528</th>\n",
       "      <th>20529</th>\n",
       "      <th>20530</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <td>A1BG.1</td>\n",
       "      <td>A1CF.29974</td>\n",
       "      <td>A2BP1.54715</td>\n",
       "      <td>A2LD1.87769</td>\n",
       "      <td>A2ML1.144568</td>\n",
       "      <td>A2M.2</td>\n",
       "      <td>A4GALT.53947</td>\n",
       "      <td>A4GNT.51146</td>\n",
       "      <td>AAA1.404744</td>\n",
       "      <td>AAAS.8086</td>\n",
       "      <td>...</td>\n",
       "      <td>ZUFSP.221302</td>\n",
       "      <td>ZW10.9183</td>\n",
       "      <td>ZWILCH.55055</td>\n",
       "      <td>ZWINT.11130</td>\n",
       "      <td>ZYG11A.440590</td>\n",
       "      <td>ZYG11B.79699</td>\n",
       "      <td>ZZEF1.23140</td>\n",
       "      <td>ZZZ3.26009</td>\n",
       "      <td>psiTPTE22.387590</td>\n",
       "      <td>tAKR.389932</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA.3N.A9WB.06</th>\n",
       "      <td>0.705057</td>\n",
       "      <td>-0.280646</td>\n",
       "      <td>-0.413046</td>\n",
       "      <td>1.75705</td>\n",
       "      <td>0.221882</td>\n",
       "      <td>-2.15541</td>\n",
       "      <td>-1.05048</td>\n",
       "      <td>-0.991263</td>\n",
       "      <td>-0.150562</td>\n",
       "      <td>0.1785</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.146231</td>\n",
       "      <td>0.116896</td>\n",
       "      <td>0.28618</td>\n",
       "      <td>1.52303</td>\n",
       "      <td>0.973086</td>\n",
       "      <td>-0.253298</td>\n",
       "      <td>-2.23982</td>\n",
       "      <td>-0.0997414</td>\n",
       "      <td>-0.878129</td>\n",
       "      <td>-0.284935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA.3N.A9WC.06</th>\n",
       "      <td>-0.271497</td>\n",
       "      <td>-0.280646</td>\n",
       "      <td>-0.413046</td>\n",
       "      <td>1.03622</td>\n",
       "      <td>-0.711574</td>\n",
       "      <td>1.85513</td>\n",
       "      <td>0.141679</td>\n",
       "      <td>0.669905</td>\n",
       "      <td>-0.150562</td>\n",
       "      <td>-0.649788</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0501261</td>\n",
       "      <td>-0.973146</td>\n",
       "      <td>-0.0479013</td>\n",
       "      <td>-0.5587</td>\n",
       "      <td>-0.440471</td>\n",
       "      <td>-0.120323</td>\n",
       "      <td>-1.23257</td>\n",
       "      <td>0.702474</td>\n",
       "      <td>-0.521455</td>\n",
       "      <td>2.52572</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA.3N.A9WD.06</th>\n",
       "      <td>0.625527</td>\n",
       "      <td>1.58431</td>\n",
       "      <td>2.57894</td>\n",
       "      <td>0.218885</td>\n",
       "      <td>-0.711574</td>\n",
       "      <td>-0.203402</td>\n",
       "      <td>0.818677</td>\n",
       "      <td>-0.991263</td>\n",
       "      <td>-0.150562</td>\n",
       "      <td>-0.869765</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0990477</td>\n",
       "      <td>0.575522</td>\n",
       "      <td>-1.3043</td>\n",
       "      <td>-0.60213</td>\n",
       "      <td>1.6199</td>\n",
       "      <td>0.284513</td>\n",
       "      <td>-0.58714</td>\n",
       "      <td>0.0140356</td>\n",
       "      <td>2.48323</td>\n",
       "      <td>-0.284935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TCGA.BF.A1PU.01</th>\n",
       "      <td>-0.418943</td>\n",
       "      <td>-0.280646</td>\n",
       "      <td>0.832644</td>\n",
       "      <td>1.06113</td>\n",
       "      <td>0.248948</td>\n",
       "      <td>-1.12785</td>\n",
       "      <td>0.451753</td>\n",
       "      <td>0.0603083</td>\n",
       "      <td>-0.150562</td>\n",
       "      <td>-0.0433683</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0852827</td>\n",
       "      <td>-0.980146</td>\n",
       "      <td>-0.0663148</td>\n",
       "      <td>-0.091513</td>\n",
       "      <td>-0.987508</td>\n",
       "      <td>-1.01398</td>\n",
       "      <td>0.810425</td>\n",
       "      <td>-1.72589</td>\n",
       "      <td>-0.930842</td>\n",
       "      <td>-0.284935</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 19363 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    29          30           31           32            33     \\\n",
       "gene               A1BG.1  A1CF.29974  A2BP1.54715  A2LD1.87769  A2ML1.144568   \n",
       "TCGA.3N.A9WB.06  0.705057   -0.280646    -0.413046      1.75705      0.221882   \n",
       "TCGA.3N.A9WC.06 -0.271497   -0.280646    -0.413046      1.03622     -0.711574   \n",
       "TCGA.3N.A9WD.06  0.625527     1.58431      2.57894     0.218885     -0.711574   \n",
       "TCGA.BF.A1PU.01 -0.418943   -0.280646     0.832644      1.06113      0.248948   \n",
       "\n",
       "                    34            35           36           37         38     \\\n",
       "gene                A2M.2  A4GALT.53947  A4GNT.51146  AAA1.404744  AAAS.8086   \n",
       "TCGA.3N.A9WB.06  -2.15541      -1.05048    -0.991263    -0.150562     0.1785   \n",
       "TCGA.3N.A9WC.06   1.85513      0.141679     0.669905    -0.150562  -0.649788   \n",
       "TCGA.3N.A9WD.06 -0.203402      0.818677    -0.991263    -0.150562  -0.869765   \n",
       "TCGA.BF.A1PU.01  -1.12785      0.451753    0.0603083    -0.150562 -0.0433683   \n",
       "\n",
       "                 ...         20517      20518         20519        20520  \\\n",
       "gene             ...  ZUFSP.221302  ZW10.9183  ZWILCH.55055  ZWINT.11130   \n",
       "TCGA.3N.A9WB.06  ...     -0.146231   0.116896       0.28618      1.52303   \n",
       "TCGA.3N.A9WC.06  ...     0.0501261  -0.973146    -0.0479013      -0.5587   \n",
       "TCGA.3N.A9WD.06  ...     0.0990477   0.575522       -1.3043     -0.60213   \n",
       "TCGA.BF.A1PU.01  ...     0.0852827  -0.980146    -0.0663148    -0.091513   \n",
       "\n",
       "                         20524         20525        20527       20528  \\\n",
       "gene             ZYG11A.440590  ZYG11B.79699  ZZEF1.23140  ZZZ3.26009   \n",
       "TCGA.3N.A9WB.06       0.973086     -0.253298     -2.23982  -0.0997414   \n",
       "TCGA.3N.A9WC.06      -0.440471     -0.120323     -1.23257    0.702474   \n",
       "TCGA.3N.A9WD.06         1.6199      0.284513     -0.58714   0.0140356   \n",
       "TCGA.BF.A1PU.01      -0.987508      -1.01398     0.810425    -1.72589   \n",
       "\n",
       "                            20529        20530  \n",
       "gene             psiTPTE22.387590  tAKR.389932  \n",
       "TCGA.3N.A9WB.06         -0.878129    -0.284935  \n",
       "TCGA.3N.A9WC.06         -0.521455      2.52572  \n",
       "TCGA.3N.A9WD.06           2.48323    -0.284935  \n",
       "TCGA.BF.A1PU.01         -0.930842    -0.284935  \n",
       "\n",
       "[5 rows x 19363 columns]"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# sample gene data\n",
    "dataframes['gene_data'].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>micro_rna</th>\n",
       "      <th>TCGA.3N.A9WC.06</th>\n",
       "      <th>TCGA.3N.A9WD.06</th>\n",
       "      <th>TCGA.BF.A1PU.01</th>\n",
       "      <th>TCGA.BF.A1PV.01</th>\n",
       "      <th>TCGA.BF.A1PX.01</th>\n",
       "      <th>TCGA.BF.A1PZ.01</th>\n",
       "      <th>TCGA.BF.A1Q0.01</th>\n",
       "      <th>TCGA.BF.A3DJ.01</th>\n",
       "      <th>TCGA.BF.A3DL.01</th>\n",
       "      <th>...</th>\n",
       "      <th>TCGA.XV.AB01.06</th>\n",
       "      <th>TCGA.YD.A89C.06</th>\n",
       "      <th>TCGA.YD.A9TA.06</th>\n",
       "      <th>TCGA.YD.A9TB.06</th>\n",
       "      <th>TCGA.YG.AA3N.01</th>\n",
       "      <th>TCGA.YG.AA3O.06</th>\n",
       "      <th>TCGA.YG.AA3P.06</th>\n",
       "      <th>TCGA.Z2.A8RT.06</th>\n",
       "      <th>TCGA.Z2.AA3S.06</th>\n",
       "      <th>TCGA.Z2.AA3V.06</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>hsa.let.7a.1</td>\n",
       "      <td>0.237300</td>\n",
       "      <td>1.413978</td>\n",
       "      <td>1.265548</td>\n",
       "      <td>0.185478</td>\n",
       "      <td>0.094122</td>\n",
       "      <td>0.451664</td>\n",
       "      <td>-0.571239</td>\n",
       "      <td>0.519428</td>\n",
       "      <td>2.149726</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.528080</td>\n",
       "      <td>-2.190976</td>\n",
       "      <td>-0.647503</td>\n",
       "      <td>-1.113845</td>\n",
       "      <td>-0.000021</td>\n",
       "      <td>-0.112561</td>\n",
       "      <td>1.255562</td>\n",
       "      <td>-0.427824</td>\n",
       "      <td>-0.161147</td>\n",
       "      <td>1.307661</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>hsa.let.7a.2</td>\n",
       "      <td>0.222888</td>\n",
       "      <td>1.409208</td>\n",
       "      <td>1.264947</td>\n",
       "      <td>0.179001</td>\n",
       "      <td>0.110415</td>\n",
       "      <td>0.445268</td>\n",
       "      <td>-0.566822</td>\n",
       "      <td>0.545780</td>\n",
       "      <td>2.146487</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.535880</td>\n",
       "      <td>-2.168277</td>\n",
       "      <td>-0.645517</td>\n",
       "      <td>-1.087783</td>\n",
       "      <td>0.004459</td>\n",
       "      <td>-0.119957</td>\n",
       "      <td>1.251081</td>\n",
       "      <td>-0.437552</td>\n",
       "      <td>-0.159200</td>\n",
       "      <td>1.300205</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>hsa.let.7a.3</td>\n",
       "      <td>0.224394</td>\n",
       "      <td>1.418898</td>\n",
       "      <td>1.265640</td>\n",
       "      <td>0.156424</td>\n",
       "      <td>0.112991</td>\n",
       "      <td>0.443217</td>\n",
       "      <td>-0.583000</td>\n",
       "      <td>0.563999</td>\n",
       "      <td>2.142916</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.559339</td>\n",
       "      <td>-2.178939</td>\n",
       "      <td>-0.656756</td>\n",
       "      <td>-1.098432</td>\n",
       "      <td>-0.006949</td>\n",
       "      <td>-0.150725</td>\n",
       "      <td>1.272062</td>\n",
       "      <td>-0.431599</td>\n",
       "      <td>-0.102585</td>\n",
       "      <td>1.309069</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>hsa.let.7b</td>\n",
       "      <td>-0.225533</td>\n",
       "      <td>0.833381</td>\n",
       "      <td>1.430398</td>\n",
       "      <td>-0.259191</td>\n",
       "      <td>1.310078</td>\n",
       "      <td>-0.237814</td>\n",
       "      <td>-0.933001</td>\n",
       "      <td>0.950472</td>\n",
       "      <td>1.836045</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.496495</td>\n",
       "      <td>-3.250218</td>\n",
       "      <td>-0.958898</td>\n",
       "      <td>-0.349975</td>\n",
       "      <td>-0.638938</td>\n",
       "      <td>-0.291756</td>\n",
       "      <td>-0.757145</td>\n",
       "      <td>-1.016035</td>\n",
       "      <td>-3.003734</td>\n",
       "      <td>0.182476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>hsa.let.7c</td>\n",
       "      <td>-1.571869</td>\n",
       "      <td>2.179965</td>\n",
       "      <td>0.263552</td>\n",
       "      <td>0.220736</td>\n",
       "      <td>0.877972</td>\n",
       "      <td>-0.190870</td>\n",
       "      <td>-0.491599</td>\n",
       "      <td>-1.763443</td>\n",
       "      <td>0.910929</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.462939</td>\n",
       "      <td>-2.789086</td>\n",
       "      <td>-1.046594</td>\n",
       "      <td>0.552762</td>\n",
       "      <td>0.439090</td>\n",
       "      <td>-0.992765</td>\n",
       "      <td>2.218663</td>\n",
       "      <td>-1.160461</td>\n",
       "      <td>0.211481</td>\n",
       "      <td>-0.196613</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 453 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      micro_rna  TCGA.3N.A9WC.06  TCGA.3N.A9WD.06  TCGA.BF.A1PU.01  \\\n",
       "0  hsa.let.7a.1         0.237300         1.413978         1.265548   \n",
       "1  hsa.let.7a.2         0.222888         1.409208         1.264947   \n",
       "2  hsa.let.7a.3         0.224394         1.418898         1.265640   \n",
       "3    hsa.let.7b        -0.225533         0.833381         1.430398   \n",
       "4    hsa.let.7c        -1.571869         2.179965         0.263552   \n",
       "\n",
       "   TCGA.BF.A1PV.01  TCGA.BF.A1PX.01  TCGA.BF.A1PZ.01  TCGA.BF.A1Q0.01  \\\n",
       "0         0.185478         0.094122         0.451664        -0.571239   \n",
       "1         0.179001         0.110415         0.445268        -0.566822   \n",
       "2         0.156424         0.112991         0.443217        -0.583000   \n",
       "3        -0.259191         1.310078        -0.237814        -0.933001   \n",
       "4         0.220736         0.877972        -0.190870        -0.491599   \n",
       "\n",
       "   TCGA.BF.A3DJ.01  TCGA.BF.A3DL.01  ...  TCGA.XV.AB01.06  TCGA.YD.A89C.06  \\\n",
       "0         0.519428         2.149726  ...        -0.528080        -2.190976   \n",
       "1         0.545780         2.146487  ...        -0.535880        -2.168277   \n",
       "2         0.563999         2.142916  ...        -0.559339        -2.178939   \n",
       "3         0.950472         1.836045  ...        -0.496495        -3.250218   \n",
       "4        -1.763443         0.910929  ...        -1.462939        -2.789086   \n",
       "\n",
       "   TCGA.YD.A9TA.06  TCGA.YD.A9TB.06  TCGA.YG.AA3N.01  TCGA.YG.AA3O.06  \\\n",
       "0        -0.647503        -1.113845        -0.000021        -0.112561   \n",
       "1        -0.645517        -1.087783         0.004459        -0.119957   \n",
       "2        -0.656756        -1.098432        -0.006949        -0.150725   \n",
       "3        -0.958898        -0.349975        -0.638938        -0.291756   \n",
       "4        -1.046594         0.552762         0.439090        -0.992765   \n",
       "\n",
       "   TCGA.YG.AA3P.06  TCGA.Z2.A8RT.06  TCGA.Z2.AA3S.06  TCGA.Z2.AA3V.06  \n",
       "0         1.255562        -0.427824        -0.161147         1.307661  \n",
       "1         1.251081        -0.437552        -0.159200         1.300205  \n",
       "2         1.272062        -0.431599        -0.102585         1.309069  \n",
       "3        -0.757145        -1.016035        -3.003734         0.182476  \n",
       "4         2.218663        -1.160461         0.211481        -0.196613  \n",
       "\n",
       "[5 rows x 453 columns]"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# sample mirna data\n",
    "dataframes['micro_rna_data'].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ensemble_id</th>\n",
       "      <th>GTEX-111YS-0006-SM-5NQBE</th>\n",
       "      <th>GTEX-1122O-0005-SM-5O99J</th>\n",
       "      <th>GTEX-1128S-0005-SM-5P9HI</th>\n",
       "      <th>GTEX-113IC-0006-SM-5NQ9C</th>\n",
       "      <th>GTEX-113JC-0006-SM-5O997</th>\n",
       "      <th>GTEX-117XS-0005-SM-5PNU6</th>\n",
       "      <th>GTEX-117YW-0005-SM-5NQ8Z</th>\n",
       "      <th>GTEX-1192W-0005-SM-5NQBQ</th>\n",
       "      <th>GTEX-11DXW-0006-SM-5NQ7Y</th>\n",
       "      <th>...</th>\n",
       "      <th>GTEX-ZVE2-0006-SM-51MRW</th>\n",
       "      <th>GTEX-ZVP2-0005-SM-51MRK</th>\n",
       "      <th>GTEX-ZVT2-0005-SM-57WBW</th>\n",
       "      <th>GTEX-ZVT3-0006-SM-51MT9</th>\n",
       "      <th>GTEX-ZVT4-0006-SM-57WB8</th>\n",
       "      <th>GTEX-ZVTK-0006-SM-57WBK</th>\n",
       "      <th>GTEX-ZVZP-0006-SM-51MSW</th>\n",
       "      <th>GTEX-ZVZQ-0006-SM-51MR8</th>\n",
       "      <th>GTEX-ZXES-0005-SM-57WCB</th>\n",
       "      <th>GTEX-ZXG5-0005-SM-57WCN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>\"ENSG00000227232\"</td>\n",
       "      <td>-0.169609</td>\n",
       "      <td>0.054703</td>\n",
       "      <td>0.009834</td>\n",
       "      <td>0.062832</td>\n",
       "      <td>-0.111598</td>\n",
       "      <td>0.013749</td>\n",
       "      <td>0.059378</td>\n",
       "      <td>-0.103492</td>\n",
       "      <td>-0.056906</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.135461</td>\n",
       "      <td>0.031812</td>\n",
       "      <td>-0.033919</td>\n",
       "      <td>-0.083926</td>\n",
       "      <td>0.057796</td>\n",
       "      <td>-0.026684</td>\n",
       "      <td>0.076550</td>\n",
       "      <td>0.074503</td>\n",
       "      <td>-0.024322</td>\n",
       "      <td>0.008076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>\"ENSG00000238009\"</td>\n",
       "      <td>-0.268123</td>\n",
       "      <td>0.129532</td>\n",
       "      <td>-0.226318</td>\n",
       "      <td>-0.594374</td>\n",
       "      <td>-0.247985</td>\n",
       "      <td>0.440750</td>\n",
       "      <td>0.012203</td>\n",
       "      <td>0.023756</td>\n",
       "      <td>0.023010</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.233336</td>\n",
       "      <td>-0.041431</td>\n",
       "      <td>0.150914</td>\n",
       "      <td>0.142757</td>\n",
       "      <td>0.181792</td>\n",
       "      <td>-0.067478</td>\n",
       "      <td>0.196848</td>\n",
       "      <td>0.086229</td>\n",
       "      <td>-0.274816</td>\n",
       "      <td>-0.030805</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>\"ENSG00000233750\"</td>\n",
       "      <td>-0.168988</td>\n",
       "      <td>-0.183577</td>\n",
       "      <td>-0.164673</td>\n",
       "      <td>-0.587763</td>\n",
       "      <td>-0.170246</td>\n",
       "      <td>0.177095</td>\n",
       "      <td>0.092399</td>\n",
       "      <td>0.186742</td>\n",
       "      <td>0.050464</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.647793</td>\n",
       "      <td>-0.179416</td>\n",
       "      <td>-0.039154</td>\n",
       "      <td>-0.157434</td>\n",
       "      <td>-0.072724</td>\n",
       "      <td>0.008101</td>\n",
       "      <td>0.326528</td>\n",
       "      <td>-0.103061</td>\n",
       "      <td>-0.018962</td>\n",
       "      <td>-0.099040</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>\"ENSG00000237683\"</td>\n",
       "      <td>-0.015901</td>\n",
       "      <td>-0.114264</td>\n",
       "      <td>0.273116</td>\n",
       "      <td>-0.442626</td>\n",
       "      <td>-0.249558</td>\n",
       "      <td>0.040346</td>\n",
       "      <td>0.092957</td>\n",
       "      <td>0.060168</td>\n",
       "      <td>0.024587</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.200647</td>\n",
       "      <td>-0.036321</td>\n",
       "      <td>0.093315</td>\n",
       "      <td>-0.097231</td>\n",
       "      <td>0.055875</td>\n",
       "      <td>-0.226099</td>\n",
       "      <td>-0.198103</td>\n",
       "      <td>-0.151890</td>\n",
       "      <td>-0.091329</td>\n",
       "      <td>-0.023483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>\"ENSG00000268903\"</td>\n",
       "      <td>0.033084</td>\n",
       "      <td>-0.136116</td>\n",
       "      <td>0.357690</td>\n",
       "      <td>-0.209781</td>\n",
       "      <td>-1.040812</td>\n",
       "      <td>0.290923</td>\n",
       "      <td>-0.224477</td>\n",
       "      <td>-0.395459</td>\n",
       "      <td>0.080350</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.395148</td>\n",
       "      <td>0.019980</td>\n",
       "      <td>0.151770</td>\n",
       "      <td>0.050996</td>\n",
       "      <td>0.102515</td>\n",
       "      <td>-0.360847</td>\n",
       "      <td>-0.076868</td>\n",
       "      <td>0.020539</td>\n",
       "      <td>-0.067300</td>\n",
       "      <td>0.040908</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 394 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         ensemble_id  GTEX-111YS-0006-SM-5NQBE  GTEX-1122O-0005-SM-5O99J  \\\n",
       "0  \"ENSG00000227232\"                 -0.169609                  0.054703   \n",
       "1  \"ENSG00000238009\"                 -0.268123                  0.129532   \n",
       "2  \"ENSG00000233750\"                 -0.168988                 -0.183577   \n",
       "3  \"ENSG00000237683\"                 -0.015901                 -0.114264   \n",
       "4  \"ENSG00000268903\"                  0.033084                 -0.136116   \n",
       "\n",
       "   GTEX-1128S-0005-SM-5P9HI  GTEX-113IC-0006-SM-5NQ9C  \\\n",
       "0                  0.009834                  0.062832   \n",
       "1                 -0.226318                 -0.594374   \n",
       "2                 -0.164673                 -0.587763   \n",
       "3                  0.273116                 -0.442626   \n",
       "4                  0.357690                 -0.209781   \n",
       "\n",
       "   GTEX-113JC-0006-SM-5O997  GTEX-117XS-0005-SM-5PNU6  \\\n",
       "0                 -0.111598                  0.013749   \n",
       "1                 -0.247985                  0.440750   \n",
       "2                 -0.170246                  0.177095   \n",
       "3                 -0.249558                  0.040346   \n",
       "4                 -1.040812                  0.290923   \n",
       "\n",
       "   GTEX-117YW-0005-SM-5NQ8Z  GTEX-1192W-0005-SM-5NQBQ  \\\n",
       "0                  0.059378                 -0.103492   \n",
       "1                  0.012203                  0.023756   \n",
       "2                  0.092399                  0.186742   \n",
       "3                  0.092957                  0.060168   \n",
       "4                 -0.224477                 -0.395459   \n",
       "\n",
       "   GTEX-11DXW-0006-SM-5NQ7Y  ...  GTEX-ZVE2-0006-SM-51MRW  \\\n",
       "0                 -0.056906  ...                -0.135461   \n",
       "1                  0.023010  ...                -0.233336   \n",
       "2                  0.050464  ...                -0.647793   \n",
       "3                  0.024587  ...                -0.200647   \n",
       "4                  0.080350  ...                -0.395148   \n",
       "\n",
       "   GTEX-ZVP2-0005-SM-51MRK  GTEX-ZVT2-0005-SM-57WBW  GTEX-ZVT3-0006-SM-51MT9  \\\n",
       "0                 0.031812                -0.033919                -0.083926   \n",
       "1                -0.041431                 0.150914                 0.142757   \n",
       "2                -0.179416                -0.039154                -0.157434   \n",
       "3                -0.036321                 0.093315                -0.097231   \n",
       "4                 0.019980                 0.151770                 0.050996   \n",
       "\n",
       "   GTEX-ZVT4-0006-SM-57WB8  GTEX-ZVTK-0006-SM-57WBK  GTEX-ZVZP-0006-SM-51MSW  \\\n",
       "0                 0.057796                -0.026684                 0.076550   \n",
       "1                 0.181792                -0.067478                 0.196848   \n",
       "2                -0.072724                 0.008101                 0.326528   \n",
       "3                 0.055875                -0.226099                -0.198103   \n",
       "4                 0.102515                -0.360847                -0.076868   \n",
       "\n",
       "   GTEX-ZVZQ-0006-SM-51MR8  GTEX-ZXES-0005-SM-57WCB  GTEX-ZXG5-0005-SM-57WCN  \n",
       "0                 0.074503                -0.024322                 0.008076  \n",
       "1                 0.086229                -0.274816                -0.030805  \n",
       "2                -0.103061                -0.018962                -0.099040  \n",
       "3                -0.151890                -0.091329                -0.023483  \n",
       "4                 0.020539                -0.067300                 0.040908  \n",
       "\n",
       "[5 rows x 394 columns]"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dataframes['gtex_data'].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PatientID</th>\n",
       "      <th>Survival</th>\n",
       "      <th>isDead</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-3N-A9WB-06</td>\n",
       "      <td>518.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCGA-3N-A9WC-06</td>\n",
       "      <td>2022.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TCGA-3N-A9WD-06</td>\n",
       "      <td>395.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCGA-BF-A1PU-01</td>\n",
       "      <td>387.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TCGA-BF-A1PV-01</td>\n",
       "      <td>14.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         PatientID  Survival  isDead\n",
       "0  TCGA-3N-A9WB-06     518.0     1.0\n",
       "1  TCGA-3N-A9WC-06    2022.0     0.0\n",
       "2  TCGA-3N-A9WD-06     395.0     1.0\n",
       "3  TCGA-BF-A1PU-01     387.0     0.0\n",
       "4  TCGA-BF-A1PV-01      14.0     0.0"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# sample survival data\n",
    "dataframes['survival_data'].head()"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "Loading Data From http://acgt.cs.tau.ac.il/multi_omic_benchmark/download.html (clinical data -> melanoma)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [],
   "source": [
    "# load clinical data data\n",
    "\n",
    "\"\"\"\n",
    "the data has a problem in line 38, 39:\n",
    "they are series with 3 elements (total values: 105, unlike total 103 in other rows)\n",
    "we discluded this data\n",
    "\"\"\"\n",
    "\n",
    "clinical_titles = pd.read_csv('melanoma.txt', header=None, nrows=1) \n",
    "clinical_data_titles = clinical_titles.iloc[0, 0].split('\\t')\n",
    "\n",
    "clinical_data = pd.read_csv('melanoma.txt', header=None, names=clinical_data_titles, skiprows=[0, 38,39], sep='\\t') \n",
    "\n",
    "dataframes['clinical_data'] = clinical_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sampleID</th>\n",
       "      <th>_EVENT</th>\n",
       "      <th>_INTEGRATION</th>\n",
       "      <th>_OS</th>\n",
       "      <th>_OS_IND</th>\n",
       "      <th>_OS_UNIT</th>\n",
       "      <th>_PATIENT</th>\n",
       "      <th>_RFS</th>\n",
       "      <th>_RFS_IND</th>\n",
       "      <th>_RFS_UNIT</th>\n",
       "      <th>...</th>\n",
       "      <th>_GENOMIC_ID_data/public/TCGA/SKCM/miRNA_HiSeq_gene</th>\n",
       "      <th>_GENOMIC_ID_TCGA_SKCM_RPPA</th>\n",
       "      <th>_GENOMIC_ID_TCGA_SKCM_mutation_bcm_gene</th>\n",
       "      <th>_GENOMIC_ID_TCGA_SKCM_mutation_broad_gene</th>\n",
       "      <th>_GENOMIC_ID_TCGA_SKCM_gistic2</th>\n",
       "      <th>_GENOMIC_ID_TCGA_SKCM_mutation</th>\n",
       "      <th>_GENOMIC_ID_TCGA_SKCM_exp_HiSeqV2_exon</th>\n",
       "      <th>_GENOMIC_ID_TCGA_SKCM_PDMRNAseqCNV</th>\n",
       "      <th>_GENOMIC_ID_TCGA_SKCM_PDMRNAseq</th>\n",
       "      <th>_GENOMIC_ID_TCGA_SKCM_exp_HiSeqV2_percentile</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-3N-A9WB-06</td>\n",
       "      <td>1.0</td>\n",
       "      <td>TCGA-3N-A9WB-06</td>\n",
       "      <td>518.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>days</td>\n",
       "      <td>TCGA-3N-A9WB</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>days</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-3N-A9WB-06A-11D-A38G-08</td>\n",
       "      <td>TCGA-3N-A9WB-06A-11D-A38F-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4c243ea9-dfe1-42f0-a887-3c901fb38542</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4c243ea9-dfe1-42f0-a887-3c901fb38542</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCGA-3N-A9WC-06</td>\n",
       "      <td>0.0</td>\n",
       "      <td>TCGA-3N-A9WC-06</td>\n",
       "      <td>2022.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>days</td>\n",
       "      <td>TCGA-3N-A9WC</td>\n",
       "      <td>2022.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>days</td>\n",
       "      <td>...</td>\n",
       "      <td>TCGA-3N-A9WC-06</td>\n",
       "      <td>F9B0B31B-24AB-4F30-8EF8-4A4ED282EF04</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-3N-A9WC-06A-11D-A38G-08</td>\n",
       "      <td>TCGA-3N-A9WC-06A-11D-A38F-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>a64ae1f5-a189-4173-be13-903bd7637869</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>a64ae1f5-a189-4173-be13-903bd7637869</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TCGA-3N-A9WD-06</td>\n",
       "      <td>1.0</td>\n",
       "      <td>TCGA-3N-A9WD-06</td>\n",
       "      <td>395.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>days</td>\n",
       "      <td>TCGA-3N-A9WD</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>days</td>\n",
       "      <td>...</td>\n",
       "      <td>TCGA-3N-A9WD-06</td>\n",
       "      <td>6AFC3678-9499-49A6-AD7A-30F3D185B4BD</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-3N-A9WD-06A-11D-A38G-08</td>\n",
       "      <td>TCGA-3N-A9WD-06A-11D-A38F-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ac19f7cf-670b-4dcc-a26b-db0f56377231</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ac19f7cf-670b-4dcc-a26b-db0f56377231</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCGA-BF-A1PU-01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>TCGA-BF-A1PU-01</td>\n",
       "      <td>387.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>days</td>\n",
       "      <td>TCGA-BF-A1PU</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>days</td>\n",
       "      <td>...</td>\n",
       "      <td>TCGA-BF-A1PU-01</td>\n",
       "      <td>7E74CAFD-C47B-4F98-A349-BBD2AC699D2D</td>\n",
       "      <td>TCGA-BF-A1PU-01A-11D-A19A-08</td>\n",
       "      <td>TCGA-BF-A1PU-01A-11D-A19A-08</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-BF-A1PU-01A-11D-A19A-08</td>\n",
       "      <td>9ff4d588-da73-4e26-975d-2b098fe74984</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9ff4d588-da73-4e26-975d-2b098fe74984</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TCGA-BF-A1PV-01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>TCGA-BF-A1PV-01</td>\n",
       "      <td>14.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>days</td>\n",
       "      <td>TCGA-BF-A1PV</td>\n",
       "      <td>14.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>days</td>\n",
       "      <td>...</td>\n",
       "      <td>TCGA-BF-A1PV-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-BF-A1PV-01A-11D-A19A-08</td>\n",
       "      <td>TCGA-BF-A1PV-01A-11D-A19A-08</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-BF-A1PV-01A-11D-A19A-08</td>\n",
       "      <td>a57841bb-fa68-4d84-85e0-2d5e4f3ce828</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>a57841bb-fa68-4d84-85e0-2d5e4f3ce828</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 103 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          sampleID  _EVENT     _INTEGRATION     _OS  _OS_IND _OS_UNIT  \\\n",
       "0  TCGA-3N-A9WB-06     1.0  TCGA-3N-A9WB-06   518.0      1.0     days   \n",
       "1  TCGA-3N-A9WC-06     0.0  TCGA-3N-A9WC-06  2022.0      0.0     days   \n",
       "2  TCGA-3N-A9WD-06     1.0  TCGA-3N-A9WD-06   395.0      1.0     days   \n",
       "3  TCGA-BF-A1PU-01     0.0  TCGA-BF-A1PU-01   387.0      0.0     days   \n",
       "4  TCGA-BF-A1PV-01     0.0  TCGA-BF-A1PV-01    14.0      0.0     days   \n",
       "\n",
       "       _PATIENT    _RFS  _RFS_IND _RFS_UNIT  ...  \\\n",
       "0  TCGA-3N-A9WB     NaN       NaN      days  ...   \n",
       "1  TCGA-3N-A9WC  2022.0       0.0      days  ...   \n",
       "2  TCGA-3N-A9WD     NaN       NaN      days  ...   \n",
       "3  TCGA-BF-A1PU     NaN       NaN      days  ...   \n",
       "4  TCGA-BF-A1PV    14.0       0.0      days  ...   \n",
       "\n",
       "   _GENOMIC_ID_data/public/TCGA/SKCM/miRNA_HiSeq_gene  \\\n",
       "0                                                NaN    \n",
       "1                                    TCGA-3N-A9WC-06    \n",
       "2                                    TCGA-3N-A9WD-06    \n",
       "3                                    TCGA-BF-A1PU-01    \n",
       "4                                    TCGA-BF-A1PV-01    \n",
       "\n",
       "             _GENOMIC_ID_TCGA_SKCM_RPPA  \\\n",
       "0                                   NaN   \n",
       "1  F9B0B31B-24AB-4F30-8EF8-4A4ED282EF04   \n",
       "2  6AFC3678-9499-49A6-AD7A-30F3D185B4BD   \n",
       "3  7E74CAFD-C47B-4F98-A349-BBD2AC699D2D   \n",
       "4                                   NaN   \n",
       "\n",
       "  _GENOMIC_ID_TCGA_SKCM_mutation_bcm_gene  \\\n",
       "0                                     NaN   \n",
       "1                                     NaN   \n",
       "2                                     NaN   \n",
       "3            TCGA-BF-A1PU-01A-11D-A19A-08   \n",
       "4            TCGA-BF-A1PV-01A-11D-A19A-08   \n",
       "\n",
       "  _GENOMIC_ID_TCGA_SKCM_mutation_broad_gene _GENOMIC_ID_TCGA_SKCM_gistic2  \\\n",
       "0              TCGA-3N-A9WB-06A-11D-A38G-08  TCGA-3N-A9WB-06A-11D-A38F-01   \n",
       "1              TCGA-3N-A9WC-06A-11D-A38G-08  TCGA-3N-A9WC-06A-11D-A38F-01   \n",
       "2              TCGA-3N-A9WD-06A-11D-A38G-08  TCGA-3N-A9WD-06A-11D-A38F-01   \n",
       "3              TCGA-BF-A1PU-01A-11D-A19A-08                           NaN   \n",
       "4              TCGA-BF-A1PV-01A-11D-A19A-08                           NaN   \n",
       "\n",
       "  _GENOMIC_ID_TCGA_SKCM_mutation _GENOMIC_ID_TCGA_SKCM_exp_HiSeqV2_exon  \\\n",
       "0                            NaN   4c243ea9-dfe1-42f0-a887-3c901fb38542   \n",
       "1                            NaN   a64ae1f5-a189-4173-be13-903bd7637869   \n",
       "2                            NaN   ac19f7cf-670b-4dcc-a26b-db0f56377231   \n",
       "3   TCGA-BF-A1PU-01A-11D-A19A-08   9ff4d588-da73-4e26-975d-2b098fe74984   \n",
       "4   TCGA-BF-A1PV-01A-11D-A19A-08   a57841bb-fa68-4d84-85e0-2d5e4f3ce828   \n",
       "\n",
       "   _GENOMIC_ID_TCGA_SKCM_PDMRNAseqCNV _GENOMIC_ID_TCGA_SKCM_PDMRNAseq  \\\n",
       "0                                 NaN                             NaN   \n",
       "1                                 NaN                             NaN   \n",
       "2                                 NaN                             NaN   \n",
       "3                                 NaN                             NaN   \n",
       "4                                 NaN                             NaN   \n",
       "\n",
       "  _GENOMIC_ID_TCGA_SKCM_exp_HiSeqV2_percentile  \n",
       "0         4c243ea9-dfe1-42f0-a887-3c901fb38542  \n",
       "1         a64ae1f5-a189-4173-be13-903bd7637869  \n",
       "2         ac19f7cf-670b-4dcc-a26b-db0f56377231  \n",
       "3         9ff4d588-da73-4e26-975d-2b098fe74984  \n",
       "4         a57841bb-fa68-4d84-85e0-2d5e4f3ce828  \n",
       "\n",
       "[5 rows x 103 columns]"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#sample clinical data\n",
    "dataframes['clinical_data'].head()"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "Clustering\n",
    "option 1: cluster all features\n",
    "option 2: remove features with low variance from dataset as part of normalization"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Option 1: cluster all features"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(145, 453)"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# dealing with nulls\n",
    "#dataframes['micro_rna_data']['TCGA.BF.A1PV.01'].isnull\n",
    "\n",
    "dataframes['micro_rna_data'].loc[dataframes['micro_rna_data']['TCGA.BF.A1PV.01'].isnull()].shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "Input contains NaN, infinity or a value too large for dtype('float64').",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-77-d1e0b085d34c>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0mX\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdataframes\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'gene_data'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0miloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 5\u001b[0;31m \u001b[0mclustering\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mSpectralClustering\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mn_clusters\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m2\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0massign_labels\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"discretize\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrandom_state\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mX\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      6\u001b[0m \u001b[0mclustering\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/anaconda3/lib/python3.6/site-packages/sklearn/cluster/spectral.py\u001b[0m in \u001b[0;36mfit\u001b[0;34m(self, X, y)\u001b[0m\n\u001b[1;32m    458\u001b[0m         \"\"\"\n\u001b[1;32m    459\u001b[0m         X = check_array(X, accept_sparse=['csr', 'csc', 'coo'],\n\u001b[0;32m--> 460\u001b[0;31m                         dtype=np.float64, ensure_min_samples=2)\n\u001b[0m\u001b[1;32m    461\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mX\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0mX\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0maffinity\u001b[0m \u001b[0;34m!=\u001b[0m \u001b[0;34m\"precomputed\"\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    462\u001b[0m             warnings.warn(\"The spectral clustering API has changed. ``fit``\"\n",
      "\u001b[0;32m/anaconda3/lib/python3.6/site-packages/sklearn/utils/validation.py\u001b[0m in \u001b[0;36mcheck_array\u001b[0;34m(array, accept_sparse, accept_large_sparse, dtype, order, copy, force_all_finite, ensure_2d, allow_nd, ensure_min_samples, ensure_min_features, warn_on_dtype, estimator)\u001b[0m\n\u001b[1;32m    540\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mforce_all_finite\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    541\u001b[0m             _assert_all_finite(array,\n\u001b[0;32m--> 542\u001b[0;31m                                allow_nan=force_all_finite == 'allow-nan')\n\u001b[0m\u001b[1;32m    543\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    544\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mensure_min_samples\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/anaconda3/lib/python3.6/site-packages/sklearn/utils/validation.py\u001b[0m in \u001b[0;36m_assert_all_finite\u001b[0;34m(X, allow_nan)\u001b[0m\n\u001b[1;32m     54\u001b[0m                 not allow_nan and not np.isfinite(X).all()):\n\u001b[1;32m     55\u001b[0m             \u001b[0mtype_err\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m'infinity'\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0mallow_nan\u001b[0m \u001b[0;32melse\u001b[0m \u001b[0;34m'NaN, infinity'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 56\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mValueError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmsg_err\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtype_err\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mX\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     57\u001b[0m     \u001b[0;31m# for object dtype data, we only check for NaNs (GH-13254)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     58\u001b[0m     \u001b[0;32melif\u001b[0m \u001b[0mX\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdtype\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'object'\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mallow_nan\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mValueError\u001b[0m: Input contains NaN, infinity or a value too large for dtype('float64')."
     ]
    }
   ],
   "source": [
    "# spectral clustering\n",
    "from sklearn.cluster import SpectralClustering\n",
    "\n",
    "X = dataframes['micro_rna_data'].iloc[:, 1:]\n",
    "clustering = SpectralClustering(n_clusters=2, assign_labels=\"discretize\", random_state=0).fit(X)\n",
    "clustering"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
